File: training-data/pascal/bubble_sort.pas
program BubbleSortDemo;

var
  data: array[0..5] of Integer = (5, 2, 9, 1, 5, 6);
  i, j, temp: Integer;
begin
  for i := 0 to 4 do
    for j := 0 to 4 - i do
      if data[j] > data[j + 1] then
      begin
        temp := data[j];
        data[j] := data[j + 1];
        data[j + 1] := temp;
      end;
  for i := 0 to 5 do
    Write(data[i], ' ');
  WriteLn;
end.

File: training-data/fortran/bubble_sort.f90
program bubble_sort
    implicit none
    integer :: arr(6) = [5, 2, 9, 1, 5, 6]
    integer :: i, j, tmp, n

    n = size(arr)
    do i = 1, n - 1
        do j = 1, n - i
            if (arr(j) > arr(j + 1)) then
                tmp = arr(j)
                arr(j) = arr(j + 1)
                arr(j + 1) = tmp
            end if
        end do
    end do
    print *, arr
end program bubble_sort
